plugins/fastem_roa_import_export.py

```diff
@@ -0,0 +1,341 @@
+        if not self._app:
+            self._app = wx.GetApp()
+        self.initialized = True
+
+    def show(self, project_list_controller=None):
+        """
+        Creates the import export feature frame and raises it if neccessary.
+
+        :param pos: The default position to show the frame at.
+        :param size: The default size of the frame.
+        :param app: A reference to the class App object.
+        :param project_list_controller: (FastEMProjectListController) class which creates/removes new
+            FastEM projects.
+
+        """
+        if not self.initialized:
+            self.init()
+
+        parent = self._app.GetTopWindow()
+        if not self._frame:
+            self._frame = ImportExportFrame(
+                parent=parent,
+                pos=self._pos,
+                size=self._size,
+                project_list_controller=project_list_controller,
+            )
+
+        self._frame.Show()
+        if self._frame.IsIconized():
+            self._frame.Iconize(False)
+        self._frame.Raise()
+
+
+class ImportExportFrame(wx.Frame):
+    """
+    This class is the frame that holds the Import Export ROAs tool. The toolbar is managed here.
+
+    :param project_list_controller: (FastEMProjectListController) class which creates/removes new
+            FastEM projects.
+
+    """
+
+    def __init__(self, title="Import Export ROAs Tool", project_list_controller=None, *args, **kw):
+        kw["title"] = title
+        wx.Frame.__init__(self, *args, **kw)
+
+        self._project_list_controller = project_list_controller
+
+        self.make_tool_bar()
+
+        self.Bind(wx.EVT_CLOSE, self.on_close)
+        if self.Parent:
+            tlw = self.Parent.GetTopLevelParent()
+            tlw.Bind(wx.EVT_CLOSE, self.on_close)
+
+    def make_tool_bar(self):
+        """Create the Import Export ROAs toolbar."""
+        tbar = self.CreateToolBar(wx.TB_HORIZONTAL | wx.TB_FLAT | wx.TB_TEXT | wx.NO_BORDER)
+        tbar.SetToolBitmapSize(wx.Size(24, 24))
+
+        import_bmp = IMPORT_IMAGE.GetBitmap()
+        export_bmp = EXPORT_IMAGE.GetBitmap()
+
+        import_tool = tbar.AddTool(-1, "Import", import_bmp, shortHelp="Import ROAs")
+        export_tool = tbar.AddTool(-1, "Export", export_bmp, shortHelp="Export ROAs")
+        tbar.Realize()
+
+        self.Bind(wx.EVT_TOOL, self.on_import, import_tool)
+        self.Bind(wx.EVT_TOOL, self.on_export, export_tool)
+
+    def on_import(self, evt):
+        """On Import ROA tool click."""
+        evt.Skip()
+        with wx.FileDialog(
+            self,
+            "Import ROAs from a CSV file",
+            wildcard="CSV files (*.csv)|*.csv",
+            style=wx.FD_OPEN | wx.FD_FILE_MUST_EXIST,
+        ) as fileDialog:
+            if fileDialog.ShowModal() == wx.ID_CANCEL:
+                return  # the user changed their mind
+
+            # Proceed loading the file chosen by the user
+            pathname = fileDialog.GetPath()
+        try:
```

Comment:
Please separate the part that asks the user for a filename from the part that read the file, from the part that take the data and applies it to the GUI. So you should have 3 methods for the import and 3 methods for the export.
That will make writing test cases easier. It's also just easier to read.